src/main/kotlin/com/jetbrains/snakecharm/lang/parser/SmkStatementParsing.kt

```diff
@@ -289,13 +306,35 @@ class SmkStatementParsing(
         referenceMarker.drop()
         return false
     }
+
+    private fun isToplevelDecoratorKeyword(): Boolean {
+        return if (myBuilder.tokenText!! in TOPLEVEL_KEYWORDS) {
```

Comment:
This method assumes that `myBuilder.tokenType == PyTokenTypes.IDENTIFIER && !scope.inPythonicSection`. Such limitation isn't evident from method name or impl which could led to misusage of the method in future by other developers. I think is better to extract whole block 
```kotlin
        if (myBuilder.tokenType == PyTokenTypes.IDENTIFIER && !scope.inPythonicSection) {
            val actualToken = SnakemakeLexer.KEYWORDS[myBuilder.tokenText!!]
            if (actualToken != null) {
                myBuilder.remapCurrentToken(actualToken)
            } else if (isToplevelDecoratorKeyword()) {
                myBuilder.remapCurrentToken(SmkTokenTypes.WORKFLOW_TOPLEVEL_DECORATOR_KEYWORD)
            }
```
to a separate method that remaps python identifier to snakemake token types + inline `isToplevelDecoratorKeyword()`  method

Reply:
https://github.com/JetBrains-Research/snakecharm/pull/405/commits/c92735ae40bedc0c60f34d6044f981bb21cab698